www/js/control.js

```diff
@@ -0,0 +1,431 @@
+            }
+            if ($scope.isIOS()) {
+                return v != "kCLLocationAccuracyBestForNavigation" && v != "kCLLocationAccuracyBest" && v != "kCLLocationAccuracyTenMeters";
+            } else if ($scope.isAndroid()) {
+                return v != "PRIORITY_HIGH_ACCURACY";
+            }
+
+        }
+    }
+    $scope.toggleLowAccuracy = function() {
+        $scope.settings.collect.new_config = JSON.parse(JSON.stringify($scope.settings.collect.config));
+        if ($scope.getLowAccuracy()) {
+            if ($scope.isIOS()) {
+                $scope.settings.collect.new_config.accuracy = $scope.settings.collect.accuracyOptions["kCLLocationAccuracyBest"];
+            } else if ($scope.isAndroid()) {
+                $scope.settings.collect.new_config.accuracy = $scope.settings.collect.accuracyOptions["PRIORITY_HIGH_ACCURACY"];
+            }
+        } else {
+            if ($scope.isIOS()) {
+                $scope.settings.collect.new_config.accuracy = $scope.settings.collect.accuracyOptions["kCLLocationAccuracyHundredMeters"];
+            } else if ($scope.isAndroid()) {
+                $scope.settings.collect.new_config.accuracy = $scope.settings.collect.accuracyOptions["PRIORITY_BALANCED_POWER_ACCURACY"];
+            }
+        }
+        ControlHelper.dataCollectionSetConfig($scope.settings.collect.new_config)
+        .then(function(){
+            console.log("setConfig Sucess");
+        }, function(err){
+            console.log("setCongif Error: " + err);
+        });
+    }
+    $scope.ionViewBackgroundClass = function() {
+        return ($scope.dark_theme)? "ion-view-background-dark" : "ion-view-background";
+    }
+    $scope.getDarkTheme = function() {
+        return $scope.dark_theme;
+    }
+    $scope.toggleDarkTheme = function() {
+        if ($scope.dark_theme) {
+            $rootScope.dark_theme = false;
+            $scope.dark_theme = false;
+            StartPrefs.setDefaultTheme(null);
+        } else {
+            $rootScope.dark_theme = true;
+            $scope.dark_theme = true;
+            StartPrefs.setDefaultTheme('dark_theme');
+        }
+        $ionicPopup.alert({template: 'Restart the app to see all changes!'})
+    }
+
+    $scope.getConnectURL = function() {
+        ControlHelper.getSettings().then(function(response) {
+            $scope.$apply(function() {
+                $scope.settings.connect.url = response.connectURL;
+                console.log(response);
+            });
+        }, function(error) {
+            console.log("While getting connect Url :" + error);
+        });
+    };
+
+    $scope.getCollectionSettings = function() {
+        var promiseList = []
+        ControlHelper.dataCollectionGetConfig().then(function(response){
+            promiseList.push(response);
+        }).then(function(){
+            ControlHelper.getAccuracyOptions().then(function(response){
+                promiseList.push(response);
+            }).then(function(){
```

Comment:
I am not sure that this is the best way to do this. With this, you are making the first call and then the second call and then dealing with the results. But the second call doesn't need any data from the first call, so it can execute in parallel with it.

To do that, I would put the two promises into the list and then use `Promise.all()` to wait for both of them to be complete and then use the results. Look at https://github.com/e-mission/cordova-usercache/blob/master/www/usercache.js#L125 for an example.